chainer/functions/normalization/decorrelated_batch_normalization.py

```diff
@@ -0,0 +1,305 @@
+from chainer import backend
+from chainer import function_node
+from chainer.utils import argument
+from chainer.utils import type_check
+
+
+def _calc_axis_and_m(x_shape, batch_size, groups):
+    m = batch_size * groups
+    spatial_ndim = len(x_shape) - 2
+    spatial_axis = tuple(range(2, 2 + spatial_ndim))
+    for i in spatial_axis:
+        m *= x_shape[i]
+    return spatial_axis, m
+
+
+class DecorrelatedBatchNormalizationFunction(function_node.FunctionNode):
+
+    def __init__(self, groups=16, eps=2e-5, mean=None, projection=None,
+                 decay=0.9):
+        self.groups = groups
+
+        self.running_mean = mean
+        self.running_projection = projection
+
+        self.eps = eps
+        self.decay = decay
+        self.axis = None
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+        x_type = in_types[0]
+        type_check.expect(
+            x_type.dtype.kind == 'f',
+            x_type.shape[1] % self.groups == 0,
+        )
+        type_check.expect(
+            x_type.ndim >= 2,
+        )
+
+    def forward(self, inputs):
+        self.retain_inputs(())
+        x = inputs[0]
+        xp = backend.get_array_module(x)
+        x_shape = x.shape
+        b, c = x_shape[:2]
+        g = self.groups
+        C = c // g
+        spatial_axis, m = _calc_axis_and_m(x_shape, b, g)
+
+        if self.running_mean is None:
```

Comment:
If they're not given by the caller, we don't need to allocate/update any running values at all.